rust_src/src/cmds.rs

```diff
@@ -413,15 +410,15 @@ fn internal_self_insert(mut c: Codepoint, n: usize) -> EmacsInt {
     }
 
     if chars_to_delete > 0 {
-        let mc = if current_buffer.multibyte_characters_enabled() && single_byte_charp(c) {
-            unibyte_to_char(c)
+        let mc = if current_buffer.multibyte_characters_enabled() && c.is_single_byte() {
+            c.unibyte_to_char()
         } else {
             c
         };
         let mut string = unsafe { Fmake_string(n.into(), mc.into(), Qnil) };
         if spaces_to_insert > 0 {
             let tem =
-                unsafe { Fmake_string(spaces_to_insert.into(), (' ' as Codepoint).into(), Qnil) };
+                unsafe { Fmake_string(spaces_to_insert.into(), Codepoint::from(' ').into(), Qnil) };
```

Comment:
Is there a way to avoid the double conversion?

Reply:
`b' '.into()` would work